lib/Request.js

```diff
@@ -276,7 +269,14 @@ class Request {
       } else {
 
         const outputBucket = this.target.bucket
-        const outputKey = key.replace(prefix, this.target.prefix)
+        let outputKey = key.replace(prefix, this.target.prefix)
+
+        // Rename if rename function exists
+        if (this.target.rename) {
+          outputKey = this.target.rename(outputKey).then(() => {
```

Comment:
changing to just `outputKey = this.target.rename` and `targetKey = this.target.rename(targetKey)`